pep-XXXX.rst

```diff
@@ -0,0 +1,492 @@
+
+This is discussed in more detail in the Runtime Behavior section.
+
+
+Reference Implementation
+========================
+
+We have a working implementation of the AST and Grammar [#callable-type-syntax--shorthand]_ with tests verifying that the grammar proposed here has the desired behaviors.
+
+There is no runtime implementation yet. At a high level we are committed to the following by backward compatibility:
+- We will need new object types for both the callable type and concatenation type, tentatively defined in C and exposed as ``types.CallableType`` and ``types.CallableConcatenateType`` in a manner similar to ``types.UnionType``.
+- The new types must support existing ``typing.Callable`` and ``typing.Concatenate`` runtime APIs almost exactly:
+  - The ``__repr__`` methods will differ and display the new builtin syntax
+  - But the ``__args__`` and ``__parameters__`` fields must behave the same
+  - And the indexing operation - which returns a new type object with concrete types substituted for various entries in ``__parameters__``, must also be the same.
+
+We will return to more details of the runtime behavior, which remain open to discussion other than backward compatibility, in the Open Issues section below.
+
+
+Open Issues
+===========
+
+Details of the Runtime API
+--------------------------
+
+The new runtime objects to which this syntax evaluates will remain backward-compatible with the ``typing.Callable`` and ``typing.Concatenate`` types they replace, other than details like ``__repr__`` where some behavior change makes sense.
+
+But we also believe that we should have a new runtime API with more structured data access, since:
+- Callable types have a more complicated shape than other generics, especially given the behavior when using ``...`` and ``typing.Concatenate``
+- In the future we might want to add more features, such as support for named and optional arguments, that would be even more difficult to describe well using only ``__args__`` and ``__parameters___``.
+
+Our tentative plan is to define enough new builtins for the runtime data to mirror the shape of the AST, but other options are also possible. See [#runtime-behavior-specification]_ for a detailed description of the current plan and a place to discuss other ideas.
+
+Once the runtime behavior is fully defined we will add a complete evaluation model and description of behavior to this PEP.
+
+Optimizing ``SyntaxError`` messages
+-----------------------------------
+
+The current reference implementation has a fully-functional parser and all edge cases presented here have been tested.
+
+But there are some known cases where the errors are not as informative as we would like. For example, because ``(int, ...) -> bool`` is illegal but ``(int, ...)`` is a valid tuple, we currently produce a syntax error flagging the ``->`` as the problem even though the real cause of the error is using ``...`` as an argument type.
```

Comment:
```suggestion
However, there are some known cases where the errors are not as informative as we would like. For example, because ``(int, ...) -> bool`` is illegal but ``(int, ...)`` is a valid tuple, we currently produce a syntax error flagging the ``->`` as the problem even though the real cause of the error is using ``...`` as an argument type.
```

Reply:
I think I'll skip this one for now, I had it drilled into me in high school that "But" is okay and "However" is bad, can't overcome that training :)